Test/Invoke-WhiskeyNpmConfig.Tests.ps1

```diff
@@ -0,0 +1,193 @@
+    if( (Test-Path -Path $projectNpmrcPath -PathType Leaf) )
+    {
+        Remove-Item -Path $projectNpmrcPath
+    }
+}
+
+function ThenConfigNotSetAtProjectLevel
+{
+    It ('should not set project-level .npmrc') {
+        Join-Path -Path $TestDrive.FullName -ChildPath '.npmrc' | Should -Not -Exist
+    }
+}
+
+function ThenConfigSetAtGlobalLevel
+{
+    It ('should set config in the user''s .npmrc file') {
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -contains '-g' }
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -notcontains '-userconfig' }
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -notcontains '.npmrc' }
+    }
+}
+
+function ThenConfigSetAtUserLevel
+{
+    It ('should set config in the global .npmrc file') {
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -notcontains '-g' }
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -notcontains '-userconfig' }
+        Assert-MockCalled -CommandName 'Invoke-WhiskeyNpmCommand' -ModuleName 'Whiskey' -ParameterFilter { $ArgumentList -notcontains '.npmrc' }
+    }
+}
+
+function ThenNpmrcDoesNotExist
+{
+    It ('should set no configuration') {
+        Join-Path -Path $TestDrive.FullName -ChildPath '.npmrc' | Should -Not -Exist
+    }
+}
+
+function ThenNpmrcIs
+{
+    param(
+        $ExpectedContent
+    )
+
+    It ('should set configuration') {
+        GEt-Content -Path (Join-Path -Path $TestDrive.FullName -ChildPath '.npmrc') -Raw | Should -Be $ExpectedContent
+    }
+}
+
+function ThenTaskFails
+{
+    param(
+        $Message
+    )
+
+    It ('should throw an exception') {
+        $threwException | Should -Be $true
+        $Global:Error | Should -Match $Message
+    }
+}
+
+function WhenRunningTask
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $Global:Error.Clear()
+
+    $context = New-WhiskeyTestContext -ForDeveloper
+    $parameter = @{ }
+    if( $configuration )
+    {
+        $parameter['Configuration'] = $configuration
+    }
+
+    if( $scope )
+    {
+        $parameter['Scope'] = $scope
+    }
+
+    try
+    {
+        Invoke-WhiskeyTask -TaskContext $context -Name 'NpmConfig' -Parameter $parameter
+    }
+    catch
+    {
+        $script:threwException = $true
+        Write-Error -ErrorRecord $_
+    }
+}
+
+# All tests are in one Describe because it takes 15 seconds or so to setup Node. With one describe, we only pay that cost once instead of in every test.
+Describe 'NpmConfig' {
+
+    try
+    {
+        Install-Node
+
+        Init
+        GivenConfiguration @{ 'key1' = 'value1' ; 'key2' = 'value2' }
+        WhenRunningTask

```

Comment:
Unfortunately, Pester clears out the test drive in each Context, so it would double or triple the test time as each context would have to re-copy Node into the test drive. The single describe block is a test duration optimization.